In [8]:
from google.colab import drive

drive.mount("/content/drive")

Mounted at /content/drive


# 0. PROJECT SETUP & TEAM CODE SYNC

This notebook is the master integration environment.

Team members develop individual modules and push their final code to GitHub.
The master Colab pulls the latest version from GitHub.

Final execution and submission generation will be done from this notebook.

In [31]:
# ==========================================
# GITHUB TEAM REPOSITORY SYNC
# ==========================================

import os

REPO_PATH = "/content/Amazon_ML_Hackathon"
REPO_URL = "https://github.com/alexkingleytuluri/Amazon_ML_Hackathon.git"

if not os.path.exists(os.path.join(REPO_PATH, ".git")):
    !git clone {REPO_URL} {REPO_PATH}
else:
    !git -C {REPO_PATH} pull origin main

print("✅ GitHub repository synced successfully.")

From https://github.com/alexkingleytuluri/Amazon_ML_Hackathon
 * branch            main       -> FETCH_HEAD
Already up to date.
✅ GitHub repository synced successfully.


In [32]:
# ==========================================
# CHECK TEAM REPOSITORY
# ==========================================

print("Files currently in GitHub repository:")

for file in os.listdir(REPO_PATH):
    print(" -", file)

Files currently in GitHub repository:
 - .git
 - README.md
 - name_cleaning_1.ipynb


In [33]:
# ==========================================
# CREATE TEAM CODE FOLDERS
# ==========================================

os.makedirs(os.path.join(REPO_PATH, "src"), exist_ok=True)
os.makedirs(os.path.join(REPO_PATH, "notebooks"), exist_ok=True)

print("✅ Project folders ready.")

✅ Project folders ready.


In [34]:
print("Repository structure:")

for item in sorted(os.listdir(REPO_PATH)):
    print(" -", item)

Repository structure:
 - .git
 - README.md
 - name_cleaning_1.ipynb
 - notebooks
 - src


In [9]:
# ==========================================
# PROJECT PATHS
# ==========================================

import os
import pandas as pd

BASE_PATH = "/content/drive/MyDrive/ML_Hackathon/student_resource"

TRAIN_PATH = os.path.join(BASE_PATH, "dataset", "train")
TEST_PATH = os.path.join(BASE_PATH, "dataset", "test")
UTILS_PATH = os.path.join(BASE_PATH, "utils")
OUTPUT_PATH = os.path.join(BASE_PATH, "output")

os.makedirs(OUTPUT_PATH, exist_ok=True)

print("BASE_PATH :", BASE_PATH)
print("TRAIN_PATH:", TRAIN_PATH)
print("TEST_PATH :", TEST_PATH)
print("UTILS_PATH:", UTILS_PATH)
print("OUTPUT_PATH:", OUTPUT_PATH)

BASE_PATH : /content/drive/MyDrive/ML_Hackathon/student_resource
TRAIN_PATH: /content/drive/MyDrive/ML_Hackathon/student_resource/dataset/train
TEST_PATH : /content/drive/MyDrive/ML_Hackathon/student_resource/dataset/test
UTILS_PATH: /content/drive/MyDrive/ML_Hackathon/student_resource/utils
OUTPUT_PATH: /content/drive/MyDrive/ML_Hackathon/student_resource/output


In [10]:
# ==========================================
# VERIFY DATASET
# ==========================================

print("TRAIN FILES:")
for file in sorted(os.listdir(TRAIN_PATH)):
    print("  ", file)

print("\nTEST FILES:")
for file in sorted(os.listdir(TEST_PATH)):
    print("  ", file)

print("\nUTILITY FILES:")
for file in sorted(os.listdir(UTILS_PATH)):
    print("  ", file)

TRAIN FILES:
   train_ground_truth.tsv
   train_source1.tsv
   train_source2.tsv
   train_source3.tsv

TEST FILES:
   test_source1.tsv
   test_source2.tsv

UTILITY FILES:
   validate_submission.py


# 1. DATA LOADING & INITIAL INSPECTION


In [11]:
# ==========================================
# LOAD TRAINING DATA
# ==========================================

train_source1 = pd.read_csv(
    os.path.join(TRAIN_PATH, "train_source1.tsv"),
    sep="\t"
)

train_source2 = pd.read_csv(
    os.path.join(TRAIN_PATH, "train_source2.tsv"),
    sep="\t"
)

train_source3 = pd.read_csv(
    os.path.join(TRAIN_PATH, "train_source3.tsv"),
    sep="\t"
)

train_ground_truth = pd.read_csv(
    os.path.join(TRAIN_PATH, "train_ground_truth.tsv"),
    sep="\t"
)

print("Training data loaded successfully.")

Training data loaded successfully.


In [12]:
# ==========================================
# TEST DATA LOAD
# ==========================================

train_source1 = pd.read_csv(
    os.path.join(TRAIN_PATH, "train_source1.tsv"),
    sep="\t"
)

print("Shape:", train_source1.shape)
display(train_source1.head())

Shape: (2206821, 4)


,entity_id,business_name,business_address,country
0,S1-925783039,Orelee's Barbershop,"1795 Westchester Drive, High Point, NC",US
1,S1-773889195,Prime Money,"17560 Ellis Road, Tahlequah, OK",US
2,S1-377745466,B+ Retail Inc,"1712 Montebello Avenue, Phoenix, AZ",US
3,S1-133037285,Christ Chapel,"2100 Cameron Drive, Unit APARTMENT G, Dundalk, MD",US
4,S1-755362802,Prabhav Business Center,"797, Lake Town Block A, Kolkata, Howrah, West ...",India


In [13]:
import psutil

ram = psutil.virtual_memory()

print(f"Total RAM     : {ram.total / (1024**3):.2f} GB")
print(f"Available RAM : {ram.available / (1024**3):.2f} GB")
print(f"Used RAM      : {ram.used / (1024**3):.2f} GB")

Total RAM     : 12.67 GB
Available RAM : 7.56 GB
Used RAM      : 4.80 GB


In [14]:
# ==========================================
# LOAD GROUND TRUTH
# ==========================================

train_ground_truth = pd.read_csv(
    os.path.join(TRAIN_PATH, "train_ground_truth.tsv"),
    sep="\t"
)

print("Shape:", train_ground_truth.shape)
display(train_ground_truth.head())
print("\nColumns:")
print(train_ground_truth.columns.tolist())

Shape: (2206821, 2)


,source1_entity_id,matched_entity_ids
0,S1-965667,"S2-681193310,S2-743505751,S3-775321672,S3-1129..."
1,S1-55344266,"S2-249013014,S2-197070651,S3-478195123,S3-3843..."
2,S1-343815751,"S2-790675320,S2-479876582,S3-878454467"
3,S1-656753428,"S2-153058913,S2-24659151,S3-679606215"
4,S1-102811957,"S2-478959098,S2-553508714,S2-625774905,S3-7280..."



Columns:
['source1_entity_id', 'matched_entity_ids']


In [ ]:
# 2. ENTITY-LEVEL TRAIN / VALIDATION SPLIT
### Owner: Member 1

In [15]:
# ==========================================
# ENTITY-LEVEL 80/20 TRAIN-VALIDATION SPLIT
# ==========================================

from sklearn.model_selection import train_test_split

# Get all Source 1 entity IDs
source1_ids = train_source1["entity_id"]

# Split Source 1 entity IDs into 80% train and 20% validation
train_ids, val_ids = train_test_split(
    source1_ids,
    test_size=0.20,
    random_state=42
)

print("Total Source 1 entities :", len(source1_ids))
print("Training entities       :", len(train_ids))
print("Validation entities     :", len(val_ids))

Total Source 1 entities : 2206821
Training entities       : 1765456
Validation entities     : 441365


In [16]:
# ==========================================
# CREATE TRAIN / VALIDATION SOURCE 1 DATA
# ==========================================

train_s1_80 = train_source1[
    train_source1["entity_id"].isin(train_ids)
].copy()

val_s1_20 = train_source1[
    train_source1["entity_id"].isin(val_ids)
].copy()

print("train_s1_80:", train_s1_80.shape)
print("val_s1_20 :", val_s1_20.shape)

train_s1_80: (1765456, 4)
val_s1_20 : (441365, 4)


In [17]:
# ==========================================
# CREATE TRAIN / VALIDATION GROUND TRUTH
# ==========================================

train_gt_80 = train_ground_truth[
    train_ground_truth["source1_entity_id"].isin(train_ids)
].copy()

val_gt_20 = train_ground_truth[
    train_ground_truth["source1_entity_id"].isin(val_ids)
].copy()

print("train_gt_80:", train_gt_80.shape)
print("val_gt_20 :", val_gt_20.shape)

train_gt_80: (1765456, 2)
val_gt_20 : (441365, 2)


In [18]:
# ==========================================
# VERIFY SPLIT
# ==========================================

assert len(train_s1_80) == len(train_gt_80)
assert len(val_s1_20) == len(val_gt_20)

assert set(train_s1_80["entity_id"]).isdisjoint(
    set(val_s1_20["entity_id"])
)

assert len(train_s1_80) + len(val_s1_20) == len(train_source1)

print("✅ Entity-level split verified successfully.")

✅ Entity-level split verified successfully.


# 3. ADDRESS CLEANING
###  Member 3

This section normalizes business addresses across Source 1, Source 2, and Source 3.

The original address columns are preserved.
Cleaned addresses are stored in separate `clean_address` columns.

In [19]:
# ==========================================
# ADDRESS DATA INSPECTION
# ==========================================

print("SOURCE 1 ADDRESS EXAMPLES")
display(
    train_source1[["business_address", "country"]]
    .sample(20, random_state=42)
)

print("\nMISSING ADDRESS COUNT")
print(train_source1["business_address"].isna().sum())

print("\nTOTAL ADDRESSES")
print(len(train_source1))

SOURCE 1 ADDRESS EXAMPLES


,business_address,country
2195840,"4850 20, Otisco, NY",US
1487051,"19034 Woodburn Road, Woodburn, IN",US
1878277,"7241 Osage Avenue, Mesa, AZ",US
993966,"Building No.4/606, Prabhul Cottage Karimbalur,...",India
565428,"4255 Charleswood Avenue, Memphis, TN",US
1645753,"1641 Virginia Lane, Hueytown, AL",US
1165724,"Sanford, ME, 31 Guillemette Street",US
363842,"12Thfloor, C, 1204, Roya Oasis, Jankalyan Naga...",India
962445,"2105 Wood Ridge Cove, Cedar Park, TX",US
964820,"A-301, New Sai Dham Chsl, Ramdev Park Road, Th...",India



MISSING ADDRESS COUNT
0

TOTAL ADDRESSES
2206821


In [20]:
# ==========================================
# ADDRESS CLEANING FUNCTION
# ==========================================

import re

def clean_address(address):
    """
    Normalize a business address for matching.
    Original address is not modified.
    """

    # Handle missing values
    if pd.isna(address):
        return ""

    # Convert to string
    address = str(address)

    # Lowercase
    address = address.lower()

    # Replace common punctuation/separators with spaces
    address = re.sub(r"[,./#\-]", " ", address)

    # Common address abbreviations
    replacements = {
        r"\brd\b": "road",
        r"\bst\b": "street",
        r"\bave\b": "avenue",
        r"\bblvd\b": "boulevard",
        r"\bln\b": "lane",
        r"\bdr\b": "drive",
        r"\bct\b": "court",
        r"\bhwy\b": "highway",
    }

    for pattern, replacement in replacements.items():
        address = re.sub(pattern, replacement, address)

    # Normalize multiple spaces
    address = re.sub(r"\s+", " ", address).strip()

    return address

In [21]:
# ==========================================
# TEST ADDRESS CLEANER
# ==========================================

test_addresses = [
    "1795 Westchester Drive, High Point, NC",
    "105 ELM ST, MORGANTON, NC",
    "1712 Montebello Avenue, Phoenix, AZ",
    "KH NO. -570/13, NEW DELHI, WEST DELHI, Delhi",
    None
]

for address in test_addresses:
    print("ORIGINAL:", address)
    print("CLEANED :", clean_address(address))
    print("-" * 70)

ORIGINAL: 1795 Westchester Drive, High Point, NC
CLEANED : 1795 westchester drive high point nc
----------------------------------------------------------------------
ORIGINAL: 105 ELM ST, MORGANTON, NC
CLEANED : 105 elm street morganton nc
----------------------------------------------------------------------
ORIGINAL: 1712 Montebello Avenue, Phoenix, AZ
CLEANED : 1712 montebello avenue phoenix az
----------------------------------------------------------------------
ORIGINAL: KH NO. -570/13, NEW DELHI, WEST DELHI, Delhi
CLEANED : kh no 570 13 new delhi west delhi delhi
----------------------------------------------------------------------
ORIGINAL: None
CLEANED : 
----------------------------------------------------------------------


In [22]:
# ==========================================
# CREATE CLEAN ADDRESS COLUMN
# ==========================================

train_source1["clean_address"] = train_source1["business_address"].apply(clean_address)

print("Address cleaning completed!")
print(train_source1.columns.tolist())

Address cleaning completed!
['entity_id', 'business_name', 'business_address', 'country', 'clean_address']


In [37]:
# ==========================================
# APPLY ADDRESS CLEANING TO SOURCE 2 & 3
# ==========================================

train_source2["clean_address"] = (
    train_source2["business_address"].apply(clean_address)
)

train_source3["clean_address"] = (
    train_source3["business_address"].apply(clean_address)
)

print("✅ Address cleaning completed for Source 2 and Source 3.")

✅ Address cleaning completed for Source 2 and Source 3.


In [38]:
# ==========================================
# VERIFY CLEANED DATA
# ==========================================

print("SOURCE 1 COLUMNS:")
print(train_source1.columns.tolist())

print("\nSOURCE 2 COLUMNS:")
print(train_source2.columns.tolist())

print("\nSOURCE 3 COLUMNS:")
print(train_source3.columns.tolist())

SOURCE 1 COLUMNS:
['entity_id', 'business_name', 'business_address', 'country', 'clean_address', 'business_name_clean']

SOURCE 2 COLUMNS:
['entity_id', 'business_name', 'business_address', 'country', 'business_name_clean', 'clean_address']

SOURCE 3 COLUMNS:
['entity_id', 'business_name', 'business_address', 'country', 'business_name_clean', 'clean_address']


In [39]:
# ==========================================
# SAMPLE CHECK — NAME + ADDRESS CLEANING
# ==========================================

print("SOURCE 1 SAMPLE")
display(
    train_source1[
        [
            "business_name",
            "business_name_clean",
            "business_address",
            "clean_address",
            "country"
        ]
    ].sample(10, random_state=42)
)

print("\nSOURCE 2 SAMPLE")
display(
    train_source2[
        [
            "business_name",
            "business_name_clean",
            "business_address",
            "clean_address",
            "country"
        ]
    ].sample(10, random_state=42)
)

print("\nSOURCE 3 SAMPLE")
display(
    train_source3[
        [
            "business_name",
            "business_name_clean",
            "business_address",
            "clean_address",
            "country"
        ]
    ].sample(10, random_state=42)
)

SOURCE 1 SAMPLE


,business_name,business_name_clean,business_address,clean_address,country
2195840,Pediatric Medicine PLLC,pediatric medicine pllc,"4850 20, Otisco, NY",4850 20 otisco ny,US
1487051,Fetech National Twin,fetech national twin,"19034 Woodburn Road, Woodburn, IN",19034 woodburn road woodburn in,US
1878277,General Design Innovations LLC,general design innovations llc,"7241 Osage Avenue, Mesa, AZ",7241 osage avenue mesa az,US
993966,Construction Ideaz Papers Private Limited,construction ideaz papers private limited,"Building No.4/606, Prabhul Cottage Karimbalur,...",building no 4 606 prabhul cottage karimbalur e...,India
565428,Penaloza and Bittle First Inc.,penaloza and bittle first inc,"4255 Charleswood Avenue, Memphis, TN",4255 charleswood avenue memphis tn,US
1645753,National Investments LLC,national investments llc,"1641 Virginia Lane, Hueytown, AL",1641 virginia lane hueytown al,US
1165724,Reus Idaho Company,reus idaho company,"Sanford, ME, 31 Guillemette Street",sanford me 31 guillemette street,US
363842,Creative Projects Limited,creative projects limited,"12Thfloor, C, 1204, Roya Oasis, Jankalyan Naga...",12thfloor c 1204 roya oasis jankalyan nagarmal...,India
962445,Goar Stone LLC,goar stone llc,"2105 Wood Ridge Cove, Cedar Park, TX",2105 wood ridge cove cedar park tx,US
964820,Lakshmi Consultants Private Limited,lakshmi consultants private limited,"A-301, New Sai Dham Chsl, Ramdev Park Road, Th...",a 301 new sai dham chsl ramdev park road thane...,India



SOURCE 2 SAMPLE


,business_name,business_name_clean,business_address,clean_address,country
2641297,Harbor Center,harbor center,"6309 EVANGELINE TRAIL, AUSTIN, TX",6309 evangeline trail austin tx,US
1856454,STEWARD ACE PREMIER,steward ace premier,"1029 MAPLE HILL RD, LEBANON, TN",1029 maple hill road lebanon tn,US
371269,Klapper and Lott,klapper and lott,"368 VINE STREET, TOOELE, UT",368 vine street tooele ut,US
793182,EAR NOSE & THROAT CARE,ear nose throat care,"YORK RD, LUTHERVILLE, MD",york road lutherville md,US
4159519,Shield Inrfabuild Private Limited,shield inrfabuild private limited,"206 FLAT NO. LG 1 KH NO. 514, 516 SHYAM BHAWAN...",206 flat no lg 1 kh no 514 516 shyam bhawan jh...,India
2290871,#holtline,holtline,"HOUSTON, 6119 DARLINGHURST DRIVE, TX",houston 6119 darlinghurst drive tx,US
2479261,Cancer [Services],cancer services,NaN,,US
606462,Smt Hermes Private Limited Center,smt hermes private limited center,"Delhi, HOUSE NO 138, 2ND FLOOR, BLK C, PKT 2 D...",delhi house no 138 2nd floor blk c pkt 2 dda m...,India
3805434,AMARDEEP PTER PRIVATE LIMITED,amardeep pter private limited,"332NEAR TELCO SERVICE STATION RANGPURI, NEW DE...",332near telco service station rangpuri new del...,India
2960649,Impex-+-Brothers,impex brothers,"FLAT B 1801, PL-15&17, SECTOR 7, PATEL HERITAG...",flat b 1801 pl 15&17 sector 7 patel heritage k...,India



SOURCE 3 SAMPLE


,business_name,business_name_clean,business_address,clean_address,country
726818,Dr Apex Pvt Ltd Partners,dr apex pvt ltd partners,"75 Tulshibaugwale Colonysahakarnagar, Pune, MH",75 tulshibaugwale colonysahakarnagar pune mh,India
1045360,Shiva Mines Pvt Ltd,shiva mines pvt ltd,"F-17, Sector 22, Noida, Gautam Buddha Nagar, उ...",f 17 sector 22 noida gautam buddha nagar उत्तर...,India
4785624,Vanguard Métropolitan Sunshine,vanguard métropolitan sunshine,"3518-B Crestview Ln, Catoosa, Oklahoma",3518 b crestview lane catoosa oklahoma,US
2503843,Vanguard Apoge-Inc,vanguard apoge inc,"2150 350, Greencastle, Indiana",2150 350 greencastle indiana,US
3149825,रियल एग्रो प्राइवेट लिमिटेड,र यल एग र प र इव ट ल म ट ड,"H.no 829 D-225vivek Vihar, Delhi, दिल्ली",h no 829 d 225vivek vihar delhi दिल्ली,India
2749542,Irinovi,irinovi,"324 Laramie Ave, Chicago, Illinois",324 laramie avenue chicago illinois,US
3399114,Jarleon Worldwide Llc,jarleon worldwide llc,"353 Spruce Pine Road, Abingdon, Maryland",353 spruce pine road abingdon maryland,US
1611580,Sfoasseeg L.L.C.,sfoasseeg l l c,"1161d Brookside Ave, Evansdale, Iowa",1161d brookside avenue evansdale iowa,US
3556867,Brixhalo t/a Creative Enterprises,brixhalo t a creative enterprises,"6, Madurai Road, Trichy, Tamil Nadu",6 madurai road trichy tamil nadu,India
4340188,Nachiket Solutions Limited,nachiket solutions limited,"138 / 6, Kiran Complex Zone - Ii, M. P. Nagar ...",138 6 kiran complex zone ii m p nagar bhopal b...,India


# 5. CANDIDATE GENERATION / BLOCKING

Goal:
Reduce the number of Source 2 and Source 3 records that need detailed
comparison against each Source 1 entity.

We will use TF-IDF similarity to retrieve a limited number of candidates
from each source before applying detailed matching features.

In [40]:
# ==========================================
# 5.1 COUNTRY DISTRIBUTION
# ==========================================

print("SOURCE 1 COUNTRY DISTRIBUTION")
print(train_source1["country"].value_counts(dropna=False))

print("\n" + "="*60)

print("SOURCE 2 COUNTRY DISTRIBUTION")
print(train_source2["country"].value_counts(dropna=False))

print("\n" + "="*60)

print("SOURCE 3 COUNTRY DISTRIBUTION")
print(train_source3["country"].value_counts(dropna=False))

SOURCE 1 COUNTRY DISTRIBUTION
country
US       1323633
India     883188
Name: count, dtype: int64

SOURCE 2 COUNTRY DISTRIBUTION
country
US       3016817
India    2017799
Name: count, dtype: int64

SOURCE 3 COUNTRY DISTRIBUTION
country
US       3170056
India    2115547
Name: count, dtype: int64


In [41]:
# ==========================================
# 5.2 NAME UNIQUENESS CHECK
# ==========================================

for name, df in [
    ("Source 1", train_source1),
    ("Source 2", train_source2),
    ("Source 3", train_source3)
]:

    print("=" * 60)
    print(name)

    print("Total records :", len(df))
    print(
        "Unique cleaned names :",
        df["business_name_clean"].nunique()
    )
    print(
        "Missing/empty cleaned names :",
        (df["business_name_clean"] == "").sum()
    )

    print(
        "Unique cleaned addresses :",
        df["clean_address"].nunique()
    )

Source 1
Total records : 2206821
Unique cleaned names : 1520680
Missing/empty cleaned names : 0
Unique cleaned addresses : 2130089
Source 2
Total records : 5034616
Unique cleaned names : 4019756
Missing/empty cleaned names : 2
Unique cleaned addresses : 4096426
Source 3
Total records : 5285603
Unique cleaned names : 4273917
Missing/empty cleaned names : 13
Unique cleaned addresses : 4443754


In [23]:
display(
    train_source1[
        ["business_address", "clean_address", "country"]
    ].sample(15, random_state=42)
)

,business_address,clean_address,country
2195840,"4850 20, Otisco, NY",4850 20 otisco ny,US
1487051,"19034 Woodburn Road, Woodburn, IN",19034 woodburn road woodburn in,US
1878277,"7241 Osage Avenue, Mesa, AZ",7241 osage avenue mesa az,US
993966,"Building No.4/606, Prabhul Cottage Karimbalur,...",building no 4 606 prabhul cottage karimbalur e...,India
565428,"4255 Charleswood Avenue, Memphis, TN",4255 charleswood avenue memphis tn,US
1645753,"1641 Virginia Lane, Hueytown, AL",1641 virginia lane hueytown al,US
1165724,"Sanford, ME, 31 Guillemette Street",sanford me 31 guillemette street,US
363842,"12Thfloor, C, 1204, Roya Oasis, Jankalyan Naga...",12thfloor c 1204 roya oasis jankalyan nagarmal...,India
962445,"2105 Wood Ridge Cove, Cedar Park, TX",2105 wood ridge cove cedar park tx,US
964820,"A-301, New Sai Dham Chsl, Ramdev Park Road, Th...",a 301 new sai dham chsl ramdev park road thane...,India


## Address Cleaning — Member 3 Complete

- Inspected Source 1 addresses.
- Source 1 contains 2,206,821 addresses with no missing values.
- Created `clean_address()` normalization function.
- Preserved original `business_address`.
- Created `clean_address` as a separate column.
- Cleaner handles lowercase conversion, punctuation normalization, common address abbreviations, and whitespace normalization.
- The same function has been applied to Source 1, Source 2, and Source 3.

**Name Cleaning Member 2 Work**

In [24]:
import re
import unicodedata
import pandas as pd

def clean_business_name(text):
    # 1. Handle missing values
    if pd.isna(text):
        return ""

    # 2. Convert to string
    text = str(text)

    # 3. Unicode normalization
    text = unicodedata.normalize("NFKC", text)

    # 4. Lowercase
    text = text.lower()

    # 5. Remove apostrophes without adding a space
    text = text.replace("'", "")

    # 6. Remove domain suffixes
    # Examples:
    # abc.com     -> abc
    # abc.in      -> abc
    # abc.co.in   -> abc
    # abc.org     -> abc
    # abc.net     -> abc
    text = re.sub(
        r"\.(?:com|in|co\.in|org|net|biz|info|edu|gov)$",
        "",
        text
    )

    # 7. Remove remaining punctuation
    text = re.sub(r"[^\w\s]", " ", text)

    # 8. Normalize whitespace
    text = re.sub(r"\s+", " ", text).strip()

    return text

In [35]:
# ==========================================
# APPLY NAME CLEANING TO ALL TRAINING SOURCES
# ==========================================

train_source1["business_name_clean"] = (
    train_source1["business_name"].apply(clean_business_name)
)

train_source2["business_name_clean"] = (
    train_source2["business_name"].apply(clean_business_name)
)

train_source3["business_name_clean"] = (
    train_source3["business_name"].apply(clean_business_name)
)

print("✅ Business name cleaning completed for Source 1, Source 2 and Source 3.")

✅ Business name cleaning completed for Source 1, Source 2 and Source 3.


In [42]:
# ==========================================
# VERIFY NAME CLEANING
# ==========================================

print("Source 1:")
display(
    train_source1[
        ["business_name", "business_name_clean"]
    ].sample(10, random_state=42)
)

print("\nSource 2:")
display(
    train_source2[
        ["business_name", "business_name_clean"]
    ].sample(10, random_state=42)
)

print("\nSource 3:")
display(
    train_source3[
        ["business_name", "business_name_clean"]
    ].sample(10, random_state=42)
)

Source 1:


,business_name,business_name_clean
2195840,Pediatric Medicine PLLC,pediatric medicine pllc
1487051,Fetech National Twin,fetech national twin
1878277,General Design Innovations LLC,general design innovations llc
993966,Construction Ideaz Papers Private Limited,construction ideaz papers private limited
565428,Penaloza and Bittle First Inc.,penaloza and bittle first inc
1645753,National Investments LLC,national investments llc
1165724,Reus Idaho Company,reus idaho company
363842,Creative Projects Limited,creative projects limited
962445,Goar Stone LLC,goar stone llc
964820,Lakshmi Consultants Private Limited,lakshmi consultants private limited



Source 2:


,business_name,business_name_clean
2641297,Harbor Center,harbor center
1856454,STEWARD ACE PREMIER,steward ace premier
371269,Klapper and Lott,klapper and lott
793182,EAR NOSE & THROAT CARE,ear nose throat care
4159519,Shield Inrfabuild Private Limited,shield inrfabuild private limited
2290871,#holtline,holtline
2479261,Cancer [Services],cancer services
606462,Smt Hermes Private Limited Center,smt hermes private limited center
3805434,AMARDEEP PTER PRIVATE LIMITED,amardeep pter private limited
2960649,Impex-+-Brothers,impex brothers



Source 3:


,business_name,business_name_clean
726818,Dr Apex Pvt Ltd Partners,dr apex pvt ltd partners
1045360,Shiva Mines Pvt Ltd,shiva mines pvt ltd
4785624,Vanguard Métropolitan Sunshine,vanguard métropolitan sunshine
2503843,Vanguard Apoge-Inc,vanguard apoge inc
3149825,रियल एग्रो प्राइवेट लिमिटेड,र यल एग र प र इव ट ल म ट ड
2749542,Irinovi,irinovi
3399114,Jarleon Worldwide Llc,jarleon worldwide llc
1611580,Sfoasseeg L.L.C.,sfoasseeg l l c
3556867,Brixhalo t/a Creative Enterprises,brixhalo t a creative enterprises
4340188,Nachiket Solutions Limited,nachiket solutions limited


In [25]:
train_source1["business_name_clean"] = train_source1["business_name"].apply(clean_business_name)

train_source2["business_name_clean"] = train_source2["business_name"].apply(clean_business_name)

train_source3["business_name_clean"] = train_source3["business_name"].apply(clean_business_name)

In [29]:
import json

with open(
    "Amazon_ML_Hackathon/name_cleaning_1.ipynb",
    "r",
    encoding="utf-8"
) as f:
    member2_notebook = json.load(f)

print("Notebook loaded successfully.")
print("Number of cells:", len(member2_notebook["cells"]))

Notebook loaded successfully.
Number of cells: 5
